cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -179,10 +176,11 @@ public AppLifecycleHttpHandler(Authenticator authenticator, CConfiguration confi
                                  ProgramRuntimeService runtimeService, StoreFactory storeFactory,
                                  StreamConsumerFactory streamConsumerFactory, QueueAdmin queueAdmin,
                                  DiscoveryServiceClient discoveryServiceClient, PreferencesStore preferencesStore,
-                                 AdapterService adapterService) {
+                                 AdapterService adapterService, NamespaceService namespaceService) {
     super(authenticator);
     this.configuration = configuration;
     this.managerFactory = managerFactory;
+    this.namespaceService = namespaceService;
     this.appFabricDir = configuration.get(Constants.AppFabric.OUTPUT_DIR);
     this.archiveDir = this.appFabricDir + "/archive";
```

Comment:
I don't think we should allow users to configure "/archive" right? Its an internal CDAP thing? They can configure the base directory.

I agree though that we use it at multiple places and it should be centrally defined. I'll do it.